multicluster/controllers/multicluster/label_identity_controller.go

```diff
@@ -0,0 +1,314 @@
+		r.labelQueue.AddRateLimited(key)
+		klog.ErrorS(err, "Failed to sync ResourceExport for label identity", "label", key)
+		return true
+	}
+
+	// If no error occurs, we Forget this item so that it does not get queued again until
+	// another change happens.
+	r.labelQueue.Forget(key)
+	return true
+}
+
+// syncLabelResourceExport checks labelToPodsCache and determines whether a ResourceExport
+// needs to be created or deleted for the label identity.
+func (r *LabelIdentityReconciler) syncLabelResourceExport(label string) error {
+	r.labelMutex.RLock()
+	defer r.labelMutex.RUnlock()
+
+	if _, exists := r.labelToPodsCache[label]; exists {
+		// The queue received an event for this label, and there are Pods referring
+		// to this label. Either 1) a new label is encountered, and we need to create
+		// a ResourceExport for it, or 2) a Pod update/delete event triggered a label
+		// deletion, but is immediately followed by another Pod event triggering
+		// add for the same label, which is a quite improbable event. We can simply
+		// ignore AlreadyExists error in ResourceExport creation for the second case.
+		if err := r.createLabelIdentityResExport(ctx, label); err != nil {
+			return err
+		}
+	} else {
+		if err := r.deleteLabelIdentityResExport(ctx, label); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// createLabelIdentityResExport creates a ResourceExport for a newly added label.
+func (r *LabelIdentityReconciler) createLabelIdentityResExport(ctx context.Context, labelToAdd string) error {
+	resExportName := getResourceExportNameForLabelIdentity(r.localClusterID, labelToAdd)
+	labelResExport := r.getLabelIdentityResourceExport(r.remoteCommonArea.GetNamespace(), resExportName, labelToAdd)
+	klog.V(4).InfoS("Creating ResourceExport for label", "resourceExport", labelResExport.Name, "label", labelToAdd)
+	err := r.remoteCommonArea.Create(ctx, labelResExport, &client.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// deleteLabelIdentityResExport deletes a ResourceExport for a stale label.
+func (r *LabelIdentityReconciler) deleteLabelIdentityResExport(ctx context.Context, labelToDelete string) error {
+	labelResExport := &mcsv1alpha1.ResourceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getResourceExportNameForLabelIdentity(r.localClusterID, labelToDelete),
+			Namespace: r.remoteCommonArea.GetNamespace(),
+		},
+	}
+	klog.V(4).InfoS("Deleting ResourceExport for label", "resourceExport", labelResExport.Name, "label", labelToDelete)
+	err := r.remoteCommonArea.Delete(ctx, labelResExport, &client.DeleteOptions{})
+	return client.IgnoreNotFound(err)
+}
+
+func (r *LabelIdentityReconciler) getLabelIdentityResourceExport(resExportNamespace, name, normalizedLabel string) *mcsv1alpha1.ResourceExport {
+	return &mcsv1alpha1.ResourceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: resExportNamespace,
```

Comment:
You can remove the parameter `resExportNamespace`  and get it directly via `r.remoteCommonArea.GetNamespace()`

Reply:
Done